daemon-scheduler/pkg/deployment/deployment_worker.go

```diff
@@ -78,38 +78,28 @@ func (d deploymentWorker) UpdateInProgressDeployment(ctx context.Context,
 		return nil, nil
 	}
 
-	updatedDeployment, err := d.updateDeployment(environment, deployment)
+	taskProgress, err := d.checkDeploymentProgress(environment, deployment)
 	if err != nil {
-		return nil, errors.Wrap(err, "Error updating the deployment")
+		return nil, errors.Wrapf(err, "Error checking deployment %s progress in environment %s",
+			deployment.ID, environment.Name)
 	}
 
-	// retrieve in-progress again to make sure it has not been updated by another process
-	// TODO: wrap the in-progress check and updateDeployment in a transaction
-	deployment, err = d.deployment.GetInProgressDeployment(ctx, environmentName)
+	updatedDeployment, err := d.updateDeployment(ctx, environment, deployment, taskProgress)
 	if err != nil {
 		return nil, err
 	}
 
-	if deployment == nil || deployment.ID != updatedDeployment.ID {
-		return nil, nil
-	}
-
-	_, err = d.environment.UpdateDeployment(ctx, *environment, *updatedDeployment)
-	if err != nil {
-		return nil, errors.Wrapf(err, "Error updating the deployment %v in the environment %v",
-			*updatedDeployment, environment.Name)
-	}
-
 	return updatedDeployment, nil
 }
 
-func (d deploymentWorker) updateDeployment(environment *types.Environment,
-	deployment *types.Deployment) (*types.Deployment, error) {
+func (d deploymentWorker) checkDeploymentProgress(environment *types.Environment,
+	deployment *types.Deployment) (*ecs.DescribeTasksOutput, error) {
```

Comment:
Can we also add a todo here to replace these types to internal types later then? I'm guessing we wont want to use CSS's types either..